content/src/content/jcr_root/apps/core/wcm/components/commons/datalayer/clientlibs/js/datalayer.js

```diff
@@ -0,0 +1,218 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2019 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+(function() {
+    "use strict";
+
+    /* eslint no-console: "off" */
+
+    var CHANGE_EVENT = "datalayer:change";
+
+    // Initializes the data layer
+    function init() {
+        window.dataLayer = window.dataLayer || [];
+        window.dataLayer.state = {};
+        window.dataLayer._listeners = [];
+        populateDataLayer();
+        handleEvents(window.dataLayer);
+        overridePush();
+        populateDataLayerAfterOverride();
+        console.log("data layer script initialized");
```

Comment:
should we write the message on console in the final version of the code?

Reply:
This pull request is a proof of concept. The logs only simulate the behavior and will be removed when the concept is validated.